react/components/MaybeContext.tsx

```diff
@@ -8,23 +8,39 @@ interface Props {
   query?: any
 }
 
-export default class MaybeContext extends PureComponent<Props & RenderContextProps> {
+export default class MaybeContext extends PureComponent<
+  Props & RenderContextProps
+> {
   public render() {
-    const {children, runtime, nestedPage, query, params} = this.props
-    const {context, props: pageProps} = runtime.extensions[nestedPage]
-    const contextComponent = context && context.component
+    const { children, runtime, nestedPage, query, params } = this.props
+    const { context, props: pageProps } = runtime.extensions[nestedPage]
     const pageContextProps = pageProps && pageProps.context
 
-    const props = contextComponent && {
-      ...pageContextProps,
-      nextTreePath: nestedPage,
-      params,
-      query,
-      ...context!.props
+    let contextComponent = context && context.component
+    let props
+
+    if (context) {
+      contextComponent = context.component
```

Comment:
Why `contextComponent` is being assigned twice?

Reply:
I removed the first assignment